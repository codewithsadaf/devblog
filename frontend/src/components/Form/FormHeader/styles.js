//imports
import styled, { css } from "styled-components";
//components styles
import { TextComponent } from "../../Layout/TextComponent";

export const Container = styled.div`
    ${({ theme }) => css`
        margin-bottom: 1em;
    `}
`;

export const Text = styled(TextComponent)`
    margin-top: 0.3em;
`;
