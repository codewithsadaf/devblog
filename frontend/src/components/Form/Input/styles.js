//imports
import styled, { css } from "styled-components";

export const FormControl = styled.div`
    display: flex;
    flex-direction: column;
    margin-bottom: 1em;
`;

export const LabelComponent = styled.label`
    display: block;
    margin-bottom: 0.3em;
    font-weight: 700;
`;

export const InputComponent = styled.input`
    ${({ theme }) => css`
        padding: 7px 0;
        outline: none;
        border: none;
        border-bottom: 1px solid ${theme.colors.gray};
        background-color: transparent;
        font-size: ${theme.font.sizes.default};
        transition: ${theme.transition.tran_05};

        &:focus {
            border-bottom: 1px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `}
`;
