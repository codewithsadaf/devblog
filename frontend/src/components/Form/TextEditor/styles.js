//imports
import styled from "styled-components";

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    margin: 1em 0;
`;

export const Label = styled.span`
    display: block;
    margin-bottom: 0.3em;
    font-weight: 700;
`;
