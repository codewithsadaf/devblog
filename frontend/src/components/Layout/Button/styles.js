//imports
import styled, { css } from "styled-components";

const handleButtonVariant = {
    default: (theme) => css`
        background: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: transparent;
            color: ${theme.colors.primaryColor};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,
    outline: (theme) => css`
        background: transparent;
        color: ${theme.colors.primaryColor};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,

    success: (theme) => css`
        background: ${theme.colors.success};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.success};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,

    danger: (theme) => css`
        background: ${theme.colors.danger};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.danger};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,
};

const handleUppercase = () => css`
    text-transform: uppercase;
`;

const handleMaxWidth = () => css`
    width: 100%;
    padding: 1em 0;
`;

const handleSmallStyles = (theme) => css`
    padding: 7px 30px;
    border-radius: 4px;
    font-size: ${theme.font.sizes.small};
    border: 1px solid;

    &:hover {
        border: 1px solid;
    }
`;

export const ButtonContainer = styled.button`
    ${({ theme, buttonVariant, uppercase, maxWidth, isSmall }) => css`
        padding: 0.5em 1.8em;
        font-size: 1em;
        border: none;
        font-weight: bold;
        cursor: pointer;
        ${handleButtonVariant[buttonVariant](theme)};
        ${uppercase && handleUppercase()};
        ${maxWidth && handleMaxWidth()};
        ${isSmall && handleSmallStyles(theme)};
    `}
`;
