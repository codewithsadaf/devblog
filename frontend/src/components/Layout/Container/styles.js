//imports
import styled from "styled-components";

export const Container = styled.div`
    min-height: 60vh;
    margin-bottom: 5em;
    padding: 0 2em;
`;
