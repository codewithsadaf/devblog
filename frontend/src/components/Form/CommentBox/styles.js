//imports
import styled, { css } from "styled-components";
//components styles
import { InputComponent } from "../Input/styles";
import { ButtonContainer as Button } from "../../Layout/Button/styles";

export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;

    & ${Button} {
        padding: 12px 16px;
    }
`;

export const CommentInput = styled(InputComponent)`
    ${({ theme }) => css`
        border: 1px solid ${theme.colors.gray};
        transition: ${theme.transition.tran_03};
        padding: 12px 7px;
        width: 90%;
        border-radius: 5px;
        font-size: 1em;
        margin-right: 7px;

        &:focus {
            border: 1px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `}
`;
