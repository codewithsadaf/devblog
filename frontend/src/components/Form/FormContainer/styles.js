//imports
import styled, { css } from "styled-components";
//components styles
import { ButtonContainer as Button } from "../../../components/Layout/Button/styles";

export const Container = styled.div`
    width: 100%;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const Form = styled.form`
    ${({ theme }) => css`
        width: 450px;
        margin-top: 3em;

        & > ${Button} {
            margin-top: 1em;
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
            min-width: 0;
            max-width: 100%;
        }
    `}
`;
