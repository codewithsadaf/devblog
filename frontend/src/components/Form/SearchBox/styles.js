//imports
import styled, { css } from "styled-components";
//components styles
import { ButtonContainer as Button } from "../../Layout/Button/styles";
import { Container as Heading } from "../../Layout/Heading/styles";
import { InputComponent } from "../Input/styles";

export const Container = styled.div`
    width: 100%;
    margin: 0 auto;
    margin-bottom: 2em;

    & ${Heading} {
        margin-bottom: 0.5em;
    }
`;

export const SearchWrapper = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        width: 90%;
        margin: 0 auto;

        & ${Button} {
            width: 20%;
            padding: 7px;
            border-radius: 2px;

            @media ${theme.media.lteMedium} {
                width: 35%;
            }
        }

        @media ${theme.media.lteMedium} {
            width: 100%;
        }
    `}
`;

export const SearchInput = styled(InputComponent)`
    ${({ theme }) => css`
        width: 78%;

        @media ${theme.media.lteMedium} {
            width: 60%;
        }
    `}
`;
