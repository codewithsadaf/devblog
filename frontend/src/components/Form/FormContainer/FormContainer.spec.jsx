//imports
import { fireEvent, screen } from "@testing-library/react";
//styles
import { renderTheme } from "../../../styles/render-theme";
import { theme } from "../../../styles/theme";
//components
import { FormContainer } from "./";

describe("<FormContainer />", () => {
    let fnMock = jest.fn();

    describe("Rendering", () => {
        it("should render FormContainer", () => {
            renderTheme(
                <FormContainer onSubmitFn={fnMock}>
                    <input type="text" placeholder="example" />
                </FormContainer>
            );
            const InputElem = screen.getByPlaceholderText("example");

            expect(InputElem).toBeInTheDocument();
        });

        it("should match snapshot", () => {
            const { container } = renderTheme(
                <FormContainer onSubmitFn={fnMock}>
                    <input type="text" placeholder="example" />
                </FormContainer>
            );

            expect(container.firstChild).toMatchSnapshot();
        });
    });

    describe("Functionality", () => {
        beforeEach(() => {
            renderTheme(
                <FormContainer onSubmitFn={fnMock}>
                    <input type="text" placeholder="example" />
                    <button type="submit">example</button>
                </FormContainer>
            );
        });

        it("should call a function on form submit", () => {
            const submitButton = screen.getByRole("button", {
                name: "example",
            });
            fireEvent.submit(submitButton);

            expect(fnMock).toBeCalledTimes(1);
        });

        it("should change Form width on mobile", () => {
            const formElement = document.querySelector("form");

            expect(formElement).toHaveStyleRule("width", "100%", {
                media: theme.media.lteMedium,
            });
        });
    });
});
