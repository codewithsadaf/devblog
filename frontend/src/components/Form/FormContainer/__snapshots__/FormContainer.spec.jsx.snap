// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormContainer /> Rendering should match snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 450px;
  margin-top: 3em;
}

@media (max-width:768px) {
  .c1 {
    width: 100%;
    min-width: 0;
    max-width: 100%;
  }
}

<div
  class="c0"
>
  <form
    class="c1"
  >
    <input
      placeholder="example"
      type="text"
    />
  </form>
</div>
`;
